nixos/modules/services/networking/firewall.nix

```diff
@@ -366,6 +486,157 @@ in
         '';
     };
 
+    networking.firewall.defaultPolicies = mkOption {
+      type = types.submodule {
+        options = {
+          input = mkOption {
+            default = "DROP";
+            type = types.enum targetTypes;
+            description = "";
+          };
+          output = mkOption {
+            default = "ACCEPT";
+            type = types.enum targetTypes;
+            description = "";
+          };
+          forward = mkOption {
+            default = "DROP";
+            type = types.enum targetTypes;
+            description = "";
+          };
+        };
+      };
+      default = {};
+      description = ''
+        Set the default policies of the main filter chains
+      '';
+    };
+
+    networking.firewall.rules = mkOption {
```

Comment:
Nixifying the FW rules seems like a great idea (imho).
However #22586 does that as well (IIRC), i.e. we have to compare/merge them.